aten/src/ATen/GeneratorForPrivareuseone.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <c10/util/intrusive_ptr.h>
+
+namespace at {
+
+using GeneratorFuncType = std::function<at::Generator(c10::DeviceIndex)>;
+
+static std::map<std::string, GeneratorFuncType>& GetGeneratorPrivateClassMap() {
+  static std::map<std::string, GeneratorFuncType> generator_privateuse1_map;
+  return generator_privateuse1_map;
+}
+
+class Register{
+public:
+  Register(std::string str, GeneratorFuncType func) {
+    TORCH_CHECK(GetGeneratorPrivateClassMap().size() == 0,
+      "Only can register the Generator for `privateuseone` once!");
+    GetGeneratorPrivateClassMap().insert(std::make_pair(str, func));
```

Comment:
hmm, the macro `REGISTERGENERATORFORPRIVATEUSE1(GeneratorClassName)` implies that the user just needs to pass in a **class name**, but the naming in this constructor implies that it's a function/lambda.

I think we should prefer just passing in the class name here, and constructing the lambda directly in this constructor. We know exactly what the lambda should look like, given the class:
```
at::Generator make_custom_generator(DeviceIndex i) {
    return make_generator<CustomGeneratorClass>(i);
}
```